dev/PushNotifications/PushNotificationsLongRunningTask/NotificationListenerManager.cpp

```diff
@@ -44,6 +45,12 @@ void NotificationListenerManager::RemoveListener(std::wstring appId)
     m_notificationListeners.erase(appId);
 }
 
+bool NotificationListenerManager::HasSinkForAppId(std::wstring const& appId)
+{
+    auto it = m_notificationListeners.find(appId);
```

Comment:
Looks like we use auto lock = m_lock.lock_shared() when we check whether the listens map is empty or not.  should we also get a shared lock first in this method?